content/en/blog/2024/spring-starter-stable/index.md

```diff
@@ -114,12 +127,58 @@ per [spec](/docs/languages/sdk-configuration/general/#otel_resource_attributes))
 
-Allowing users to use Spring beans for configuration was a bit more challenging.
-We came up with a new interface,
+### Using Spring beans for advanced configuration
+
+Spring Boot users also expect to be able to use Spring beans for advanced configuration.
+The SDK auto-configuration did not know about Spring beans, so we came up with a new interface,
 [ComponentLoader](https://github.com/open-telemetry/opentelemetry-java/blob/release/v1.40.x/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/internal/ComponentLoader.java),
 that allows users to register Spring beans that will be loaded by the
-OpenTelemetry SDK, which can be used for advanced configuration like
-[dynamic auth headers](/docs/zero-code/java/spring-boot-starter/sdk-configuration/#configure-the-exporter-programmatically).
+OpenTelemetry SDK auto-configuration.
+
+The [Spring Starter implementation of the ComponentLoader interface](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java#L162-L181) 
+uses Spring's `ApplicationContext` to find all beans of a certain type.
+
+This allows users to register their own exporters, processors, and other SDK components as Spring beans.
+
+The most common use case is to create a bean that returns an 
+[AutoConfigurationCustomizerProvider](/docs/languages/java/configuration/#autoconfigurationcustomizerprovider) 
+instance - an idiomatic Spring Boot approach to customization.
+
+This example shows how to create a bean that customizes the sampler to drop spans for paths starting with `/actuator`:
+
+<!-- prettier-ignore-start -->
+<?code-excerpt "src/main/java/otel/FilterPaths.java"?>
+```java
+package otel;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.contrib.sampler.RuleBasedRoutingSampler;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+import io.opentelemetry.semconv.UrlAttributes;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class FilterPaths {
+
+  @Bean
+  public AutoConfigurationCustomizerProvider otelCustomizer() {
+    return p ->
+        p.addSamplerCustomizer(
+            (fallback, config) ->
+                RuleBasedRoutingSampler.builder(SpanKind.SERVER, fallback)
+                    .drop(UrlAttributes.URL_PATH, "^/actuator")
+                    .build());
+  }
+}
+```
+<!-- prettier-ignore-end -->
+
+You can also create your own exporter, to 
+[export spans to memory](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/1b5d47eadcef33405b3065ff57e9d96bb15d795a/smoke-tests-otel-starter/spring-smoke-testing/src/main/java/io/opentelemetry/spring/smoketest/SpringSmokeOtelConfiguration.java#L50-L63).
```

Comment:
This other configuration is also needed to have this metric exporter works: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/1b5d47eadcef33405b3065ff57e9d96bb15d795a/smoke-tests-otel-starter/spring-smoke-testing/src/main/java/io/opentelemetry/spring/smoketest/SpringSmokeOtelConfiguration.java#L50-L63 If we keep this example, we should mention the two required configurations.